File: src/main/java/com/github/hhhzzzsss/songplayer/commands/Command.java
package com.github.hhhzzzsss.songplayer.commands;

import com.mojang.brigadier.builder.LiteralArgumentBuilder;
import net.fabricmc.fabric.api.client.command.v2.FabricClientCommandSource;

abstract class Command {
    public abstract String getName();

    public abstract String getDescription();

    public abstract void buildNode(LiteralArgumentBuilder<FabricClientCommandSource> node);
}

File: src/main/java/com/github/hhhzzzsss/songplayer/commands/Command.kt
package com.github.hhhzzzsss.songplayer.commands

import com.mojang.brigadier.builder.LiteralArgumentBuilder
import net.fabricmc.fabric.api.client.command.v2.FabricClientCommandSource

interface Command {
    val name: String

    val description: String

    fun buildNode(node: LiteralArgumentBuilder<FabricClientCommandSource>)
}